### PR TITLE
Update traits.build book sections

### PR DESCRIPTION
**Pull request by ehwenk**
Many of the blank sections have now been filled in.

### CODE REVIEW

File: adding_data_brief.qmd [hunk 1]
```diff
@@ -23,6 +23,30 @@ schema <- get_schema()
...
-This chapter gives a brief overview of how to add datasets to the database. For a more detailed guide, see [adding_datasets](adding_data_long.html).
+This section gives a brief overview of how to add datasets to the database. For a more detailed guide, see [adding_datasets](adding_data_long.html).
 
-XXX add details
+It is important that all steps are followed so that our automated workflow proceeds without problems.
+
+1. Clone the `traits.build` repository from github
+2. Create a new branch in the repo, named for the new `dataset_id` in `author_year` format, e.g. `Gallagher_2014`.
+3. Create a new folder within the folder `data` with the name `dataset_id`, e.g. `Gallagher_2014`.
+4. Prepare the file `data.csv` and place it within the new folder ([details](#csv_file) here).
```
Comment (dfalster) [inline adding_data_brief.qmd:33]: Not sure this is the best link. That's just generic about csvs

File: adding_data_brief.qmd [hunk 2]
```diff
@@ -23,6 +23,30 @@ schema <- get_schema()
...
+5. Prepare the file `metadata.yml` and place it within the new folder ([details](#metadata_file) here).
+6. Add the new study into the build framework and rebuild AusTraits, by running `build_setup_pipeline()`.
+
+This step updates the file `remake.yml` with appropriate rules for the new dataset; similarly if you remove datasets, do the same. (At this stage, [remake](https://github.com/richfitz/remake) offers no looping constructs, so for now we generate the remake file using: [whisker](https://github.com/edwindj/whisker).)
```
Comment (dfalster) [inline adding_data_brief.qmd:37]: There's now 3 build options, base R, furrr, remake. I suggest new users use base. Add link to section on different pipelines

File: adding_data_brief.qmd [hunk 3]
```diff
@@ -23,6 +23,30 @@ schema <- get_schema()
...
+
+You can then rebuild AusTraits, including your dataset.
+
+7. Run tests and quality checks on the newly added dataset and correct the `data.csv` and `metadata.yml` files as necessary.
```
Comment (dfalster) [inline adding_data_brief.qmd:41]: Should this happen before building?

File: adding_data_brief.qmd [hunk 4]
```diff
@@ -23,6 +23,30 @@ schema <- get_schema()
...
+8. Generate and proofread a report on the data. In particular, check that numeric trait values fall within a logical range relative to other studies, and that individual trait observations are not unnecessarily excluded because their trait values are unsupported.
```
Comment (dfalster) [inline adding_data_brief.qmd:42]: Link to instructions?

File: adding_data_brief.qmd [hunk 5]
```diff
@@ -23,6 +23,30 @@ schema <- get_schema()
...
+9. Return to step 6 if changes are made to the `data.csv` or `metadata.yml` files.
+10. Push the GitHub branch.
+
+The best place to get started learning how to add datasets is to work through a series of 7 [tutorials](tutorial_dataset_1.html). Each introduces you to specific [traits.build functions](https://traitecoevo.github.io/traits.build/reference/index.html) designed to facilitate the addition of dataset metadata or the metadata formats for specific types of datasets. 
+
+It may also help to download one of the [existing datasets](https://github.com/traitecoevo/austraits.build/tree/master/data) to use as a template for your own files and a guide on required content.
```
Comment (dfalster) [inline adding_data_brief.qmd:48]: the template maybe easier. AusTraits is very big!

File: create_dictionary.qmd [hunk 6]
```diff
@@ -1,4 +1,11 @@
 # Creating a dictionary
 
-XXX
-a
+A core component of the traits.build database structure is a trait dictionary that clearly defines and describes each trait concept included within the trait database.
```
Comment (dfalster) [inline create_dictionary.qmd:3]: `traits.build` rather than trait.build, throughout

File: create_dictionary.qmd [hunk 7]
```diff
@@ -1,4 +1,11 @@
...
+
+Building upon the standard outlined by the ETS [Ecological Trait-data Standard](https://doi.org/10.1111/2041-210X.13288), all numeric traits must have standard units and an allowable range. Categorical traits must have a list of allowable trait values.
+
+Each trait concept must have a complete, explicit definition to eliminate confusion between trait concepts.
+
+The trait dictionary is compiled in a simple yml format, allowing trait definitions to be easily added and edited.
```
Comment (dfalster) [inline create_dictionary.qmd:9]: Link to example in AusTraits. And show sections of an example for one trait numeric and categorical trait here. Ideally read in from file.

File: create_dictionary.qmd [hunk 8]
```diff
@@ -1,4 +1,11 @@
...
+
+For AusTraits, the trait dictionary has been formalised into a standalone output, the AusTraits Plant Dictionary (APD), available in both machine-readable and humman-friendly outputs through the [w3id.org/APD](https://w3id.org/APD) namespace. The traits included in the APD have undergone a rigorous internal and external review process to ensure the trait concepts and their definitions are complete and robust.
```
Comment (dfalster) [inline create_dictionary.qmd:11]: Make clear that don't need this. LINK TO ACTUAL VERSION USED IN THE BUILD

File: database_standard.qmd [hunk 9]
```diff
@@ -16,5 +15,63 @@ library(traits.build)
 my_kable_styling <- util_kable_styling_html
 ```
 
-XXX Describe the emerhign traits.build data standard
+### Overview
+
+A single universally accepted trait database standard does not currently exist among plant ecologists, limiting our ability to merge together distinct databases.\
```
Comment (dfalster) [inline database_standard.qmd:20]: Why end with `\`

File: file_organisation.qmd [hunk 10]
```diff
@@ -90,23 +90,27 @@ config
...
 
-XXX Explan this more. Only show essential variables in table below.
+It includes all unique taxon names after typos have been corrected (through taxonomic_updates). It includes both accepted/valid taxon concepts and outdated taxonomic names. It includes taxon names indicating a taxon that can be identified to species and names that indicate name-alignment to a lower taxon rank.
 
-<!-- Each species is listed once, with links to species' identifiers provided by the Australian Plant Name Index (APNI). The file `taxon_list.csv` is added to if a study includes taxa not previously represented in AusTraits. These can be names included in either the APC/APNI, compilations of taxonomic concepts (APC) or names (APNI) for plants that are either native to or naturalised in Australia, or taxa without recognised names. -->
+For any species (& infraspecific taxon concepts) or genera that align with known taxon concepts, identifiers for these taxon concepts are included if available in a master taxon list. For instance for AusTraits, identifiers from the two vascular plants National Species Lists (NSL), the APC (Australian Plant Census) and the Australian Plant Name Index (APNI) are included.
```
Comment (dfalster) [inline file_organisation.qmd:105]: Make clear that APC is only for AusTraits. Other compilations won't use this

File: file_organisation.qmd [hunk 11]
```diff
@@ -90,23 +90,27 @@ config
...
+
+In the file, `cleaned_name` refers to the taxon name after any typos have been corrected, while `taxon_name` is the taxon name following updates to the currently accepted/valid taxon name (when available).
+
+<!-- Each species is listed once, with links to species' identifiers provided by the APNI. The file `taxon_list.csv` is added to if a study includes taxa not previously represented in AusTraits. These can be names included in either the APC/APNI, compilations of taxonomic concepts (APC) or names (APNI) for plants that are either native to or naturalised in Australia, or taxa without recognised names. -->
 
 ```{r, results='show'}
-read_csv("data/taxon_list.csv", show_col_types = FALSE) %>%
-  select(taxon_name, family, scientific_name_authorship, taxonomic_reference, cleaned_name_taxonomic_status, cleaned_scientific_name_id
+read_csv("config/taxon_list.csv", show_col_types = FALSE) %>%
```
Comment (dfalster) [inline file_organisation.qmd:112]: In this repo it's in data, not config

File: file_organisation.qmd [hunk 12]
```diff
@@ -545,6 +549,26 @@ questions:
...
+
+- Move specific categorical trait values to a second trait
+
+For instance, there are many datasets where a species-level trait measurement is repeated across many rows of data but should only be incorporated into the dataset a single time:
+
+```
+custom_R_code: `
+  data %>%
+    mutate(
+      across(c(`plant_growth_form`, `leaf_shape`), replace_duplicates_with_NA)
```
Comment (dfalster) [inline file_organisation.qmd:568]: Make clear that this function is part of AusTraits, not traits.build

**Review by dfalster — APPROVED** [review]

File: adding_data_brief.qmd (hunk 1 above)
Comment (ehwenk) [reply adding_data_brief.qmd:33]: Have removed the links here and on the next line

File: adding_data_brief.qmd (hunk 3 above)
Comment (ehwenk) [reply adding_data_brief.qmd:41]: Bit of both, so I've split tests and quality checks into two items, 1 before (tests) and 1 after (quality checks)

File: adding_data_brief.qmd (hunk 4 above)
Comment (ehwenk) [reply adding_data_brief.qmd:42]: added

File: adding_data_brief.qmd (hunk 5 above)
Comment (ehwenk) [reply adding_data_brief.qmd:48]: the template just has two very simple datasets - so I've referenced both

File: create_dictionary.qmd (hunk 6 above)
Comment (ehwenk) [reply create_dictionary.qmd:3]: confused - it is traits.build here And I've searched the entire repository and they are all traits.build

Comment (dfalster) [reply create_dictionary.qmd:3]: Sorry, my comment was suggesting to add the tick marks around traits.build, so "`traits.build`", not "traits.build"

File: database_standard.qmd (hunk 9 above)
Comment (ehwenk) [reply database_standard.qmd:20]: Forced line break - I can't figure out why some are immediately reformatted and others remain in the text.

File: file_organisation.qmd (hunk 11 above)
Comment (ehwenk) [reply file_organisation.qmd:112]: @dfalster I'm confused. In traits.build it isn't included at all. In traits.build-template and austraits.build it is in config. I thought this `file_organisation.qmd` was describing the file structure of a traits.build database, not the traits.build repo.

File: file_organisation.qmd (hunk 12 above)
Comment (ehwenk) [reply file_organisation.qmd:568]: This section is explicitly about the custom_R_code.R file in austraits.build - header is `## R/custom_R_code.R`

**Pull request merged**